### DISCUSSION TITLE
eslint is blocking any npm install, update, uninstall: Conflicting peer dependency: eslint@7.32.0

### DISCUSSION THREAD

**Discussion by frankmehlhop** [angular-eslint/angular-eslint#973] [General]
Most **updates and all install fail** and always I get the hint to **eslint**.
I tried to remove eslint (by using vs code, going to extensions, choosing ESLint and click on uninstall), 
but there are left overs and I still get error messages about eslint.

**What could be the cause for this behavior and how could I solve this problem?**

Thanks! Frank

I'll show some examples (ESLint is installed and to find as extension):

PS C:\Users\FrankMehlhopSOMACOS\source\repos\G5\Feature\Feature\ClientApp> **ng update @angular-eslint/schematics**
Using package manager: 'npm'
Collecting installed dependencies...
Found 29 dependencies.
Fetching dependency metadata from registry...
    Updating package.json with dependency @angular-eslint/builder @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin-template @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/schematics @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/template-parser @ "13.2.0" (was "13.0.1")...
  UPDATE package.json (1353 bytes)
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: eslint-plugin-jsdoc@30.7.6
npm ERR! Found: eslint@8.6.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^8.2.0" from the root project
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0" from @typescript-eslint/eslint-plugin@5.3.0
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"5.3.0" from the root project
npm ERR!   13 more (@typescript-eslint/experimental-utils, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR! node_modules/eslint-plugin-jsdoc
npm ERR!   dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR!   node_modules/eslint-plugin-jsdoc
npm ERR!     dev eslint-plugin-jsdoc@"30.7.6" from the root project

PS C:\Users\FrankMehlhopSOMACOS\source\repos\G5\Feature\Feature\ClientApp> **ng update @angular-eslint/schematics --force**
Using package manager: 'npm'
Collecting installed dependencies...
Found 0 dependencies.
Package '@angular-eslint/schematics' is not a dependency.

PS C:\Users\FrankMehlhopSOMACOS\source\repos\G5\Feature\Feature\ClientApp> **ng update @angular-eslint/schematics**
Using package manager: 'npm'
Collecting installed dependencies...
Found 29 dependencies.
Fetching dependency metadata from registry...
    Updating package.json with dependency @angular-eslint/builder @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin-template @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/schematics @ "13.2.0" (was "13.0.1")...
    Updating package.json with dependency @angular-eslint/template-parser @ "13.2.0" (was "13.0.1")...
  UPDATE package.json (1353 bytes)
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: eslint-plugin-jsdoc@30.7.6
npm ERR! Found: eslint@8.6.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^8.2.0" from the root project
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0" from @typescript-eslint/eslint-plugin@5.3.0
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"5.3.0" from the root project
npm ERR!   13 more (@typescript-eslint/experimental-utils, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR! node_modules/eslint-plugin-jsdoc
npm ERR!   dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR!   node_modules/eslint-plugin-jsdoc
npm ERR!     dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR!   dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR!   node_modules/eslint-plugin-jsdoc
npm ERR!     dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\_logs\2022-04-05T09_15_36_647Z-debug-0.log

PS C:\Users\FrankMehlhopSOMACOS\source\repos\G5\Feature\Feature\ClientApp> **npm install --save-dev @angular/cli@latest**
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: eslint-plugin-jsdoc@30.7.6
npm ERR! Found: eslint@8.6.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0" from @typescript-eslint/eslint-plugin@5.3.0
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"5.3.0" from the root project
npm ERR!   peer eslint@"*" from @typescript-eslint/experimental-utils@5.3.0
npm ERR!   node_modules/@typescript-eslint/experimental-utils
npm ERR!     @typescript-eslint/experimental-utils@"5.3.0" from @typescript-eslint/eslint-plugin@5.3.0
npm ERR!     node_modules/@typescript-eslint/eslint-plugin
npm ERR!       dev @typescript-eslint/eslint-plugin@"5.3.0" from the root project
npm ERR!   13 more (@typescript-eslint/parser, eslint-plugin-import, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR! node_modules/eslint-plugin-jsdoc
npm ERR!   dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6
npm ERR!   node_modules/eslint-plugin-jsdoc
npm ERR!     dev eslint-plugin-jsdoc@"30.7.6" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\_logs\2022-04-05T09_18_11_233Z-debug-0.log

**Comment by JamesHenry** [comment]
Hi @frankmehlhop,

I know it's not the most accessible error messaging coming from npm here but the information you need to proceed is actually in there I believe:

> peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.6

Is telling you that your current version of `eslint-plugin-jsdoc` (`30.7.6`) is trying to pull in either ESLint 6.x or ESLint 7.x but the rest of your tooling is ESLint 8.x

You will need to upgrade `eslint-plugin-jsdoc` to a version which is compatible with ESLint 8.x

**Reply by frankmehlhop** [reply]
Hi @JamesHenry, 
thank you for you fast answer!
**npm install --save-dev eslint-plugin-jsdoc --force**
... seems to work.
Thank you!

**Reply by frankmehlhop** [reply]
Sorry @JamesHenry,
it's not working.
Is there a different command to upgrade? (instead of using _npm install --save-dev eslint-plugin-jsdoc_)

Please check this:
PS C:\Users\FrankMehlhopSOMACOS\source\repos\G5\Feature\Feature\ClientApp> **npm update**
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: feature@0.0.0
npm ERR! Found: eslint@8.12.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^8.12.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^6.0.0 || ^7.0.0" from eslint-plugin-jsdoc@30.7.13
npm ERR! node_modules/eslint-plugin-jsdoc
npm ERR!   dev eslint-plugin-jsdoc@"^30.7.6" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\FrankMehlhopSOMACOS\AppData\Local\npm-cache\_logs\2022-04-05T10_03_01_516Z-debug-0.log

**Reply by JamesHenry** [reply]
You shouldn't need to run a special command:

Find `eslint-plugin-jsdoc` in your package.json and update its version number to a version compatible with ESLint 8 and then `npm install`.

In looking at the `eslint-plugin-jsdoc` repo I can see ESLint 8 support was introduced in version `37.0.0` but I imagine you'll want to just try and use whatever the latest version is
